File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<char> seen = new();
        int l = 0, r = 0, n = s.Length;
        if(n <= 0) {
            return 0;
        }
        int res = 0;
        while (r < n) { 
            if (seen.Contains(s[r])) 
            { seen.Remove(s[l]); l++; } 
            else { seen.Add(s[r]); r++; res = Math.Max(res, r - l); }}
        return res;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> seen;
        int res = 0;
        for(auto i : nums) {
            if(!seen[i]) {
                int count = seen[i-1] + 1 + seen[i+1];
                seen[i] = count;
                seen[i - seen[i-1]] = count;
                seen[i + seen[i+1]] = count;
                res = max(res, count);
            }
        }
        return res;
    }
};
